### PR TITLE
{Release} Upgrade to Azure CLI 2.54.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

**Comment by yonzhan** [comment]
Release

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,216 @@
...
+
+* `az acr artifact-streaming`: Add new group to include a subgroup operation and the commands create/show/update (#27594)
+* `az acr artifact-streaming operation`: Add new group to help the user manage the artifact streaming creation with the commands cancel/show (#27594)
+* `az acr login`: Add additional parameter validation to check registry name is valid (#27762)
+* `az acr cache`: Add optional `resource-group` param (#27605)
+* `az acr credential-set`: Add optional `resource-group` param (#27605)
+
+**AKS**
+
+* [BREAKING CHANGE] `aks create`: Make container log V2 as default and remove region dependency (#27615)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:19]:
```suggestion
* [BREAKING CHANGE] `az aks create`: Make container log V2 as default and remove region dependency (#27615)
```

File: src/azure-cli-core/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,11 @@
 Release History
 ===============
 
+2.54.0
+++++++
+* `aaz`: Support data-plane clients (#27617)
+* Support disabling instance discovery by running `az config set core.instance_discovery=false` (#27494)
```
Comment (jiasli) [inline src/azure-cli-core/HISTORY.rst:9]:
```suggestion
* [PREVIEW] Support disabling instance discovery by running `az config set core.instance_discovery=false` (#27494)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,216 @@
...
+* [BREAKING CHANGE] `az appconfig kv export`: Add validations to snapshot kv export to App Service (#27737)
+* [BREAKING CHANGE] `az appconfig kv import/export`: Add skip-features and skip-keyvault restriction for snapshots (#27714)
+* `az appconfig snapshot`: GA snapshot related features (#27738)
+
+**App Service**
+
+* [BREAKING CHANGE] `az functionapp`: Redact key value output on keys set (#27611)
+* [BREAKING CHANGE] Redact output on deployment source config (#27628)
+* [BREAKING CHANGE] `az webapp config`: Redact webapp config connection-string and storage-account values (#27629)
+* [BREAKING CHANGE] Redact tokens output on deployment source update-token (#27614)
```
Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:44]:
```suggestion
* [BREAKING CHANGE] `az functionapp/webapp deployment source update-token`: Redact tokens output on deployment source update-token (#27614)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,216 @@
...
+* [BREAKING CHANGE] `az appservice ase create`: Update the default App Service Environment to V3 (#27761)
+* `az appservice list-locations`: Add `--hyperv-workers-enabled` parameter to get regions which support hosting web apps on windows container workers (#27535)
+* `az functionapp deployment source config-zip`: Adding the deployer information to improve telemetry (#27427)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:47]:
```suggestion
* `az functionapp deployment source config-zip`: Add the deployer information to improve telemetry (#27427)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,216 @@
...
+
+**ARO**
+
+* `az aro create`: Add new `--enable-preconfigured-nsg` parameter, allowing users to enable or disable preconfigured NSGs (#27511)
+* `az aro create`: Add network contributor to the NSG resource for the cluster SP and FP SP (#27511)
+* `az aro update`: Add network contributor to the NSG resource for the cluster SP and FP SP, if not already when preconfigured NSG is enabled (#27511)
+
+**Backup**
+
+* Update help text for `check-vm` (#27569)
```
Comment (bebound) [inline src/azure-cli/HISTORY.rst:69]:
```suggestion
```

Comment (evelyn-ys) [inline src/azure-cli/HISTORY.rst:69]:
```suggestion
```

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,216 @@
...
+* `az keyvault backup/restore start`: Add `--use-managed-identity` to exempt SAS token (#27420)
+* `az keyvault key`: Add hsm platform info in response (#27780)
+
+**Monitor**
+
+* [BREAKING CHANGE] `az monitor activity-log alert create`: Change default value from resourceGroupId to subscriptionId for parameter `--scope` (#27126)
+* [BREAKING CHANGE] `az monitor metrics alert`: Change datetime output to be consistent with native response (#27328)
+* [BREAKING CHANGE] `az monitor log-analytics workspace table search-job create`: Remove `schema` wrapper for `searchResults` in api response (#26949)
+* `az monitor log-analytics workspace create`: Add `--identity-type` and `--user-assigned` arguments (#26949)
+* `az monitor log-analytics workspace update`: Add `--identity-type` and `--user-assigned` arguments (#26949)
```
Comment (jiasli) [inline src/azure-cli/HISTORY.rst:136]: L132 uses "parameter" but L135 uses "argument".

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,216 @@
...
+
+**SQL**
+
+* `az sql db create/update`: Add `--use-free-limit` and `--free-limit-exhaustion-behavior` to support free limit database (#27553)
+
+**Storage**
+
+* `az storage file/directory`: add `--disallow-trailing-dot` (#27622)
+* `az storage share list-handle/close-handle`: add `--disallow-trailing-dot` (#27622)
+* `az storage file copy start/start-batch`: add `--disallow-source-trailing-dot` (#27622)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:211]:
```suggestion
* `az storage file/directory`: Add `--disallow-trailing-dot` (#27622)
* `az storage share list-handle/close-handle`: Add `--disallow-trailing-dot` (#27622)
* `az storage file copy start/start-batch`: Add `--disallow-source-trailing-dot` (#27622)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,216 @@
...
+* `az postgres flexible-server upgrade`: Add MVU support for PG version 15 (#27773)
+
+**Role**
+
+* [BREAKING CHANGE] `az role assignment create`: `--scope` is now a required argument. (#27651)
+* [BREAKING CHANGE] `az role assignment create`: Remove `--resource-group` argument. (#27651)
+
+**Search**
+
+* Add 2023-11-01 support (#27632)
```
Comment (kairu-ms) [inline src/azure-cli/HISTORY.rst:183]:
```suggestion
* `az search service create`: Add `--semantic-search` argument. (#27632)
```

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,216 @@
...
+2.54.0
+++++++
+
+**ACR**
+
+* `az acr artifact-streaming`: Add new group to include a subgroup operation and the commands create/show/update (#27594)
+* `az acr artifact-streaming operation`: Add new group to help the user manage the artifact streaming creation with the commands cancel/show (#27594)
+* `az acr login`: Add additional parameter validation to check registry name is valid (#27762)
+* `az acr cache`: Add optional `resource-group` param (#27605)
+* `az acr credential-set`: Add optional `resource-group` param (#27605)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:15]:
```suggestion
* `az acr cache`: Add optional `--resource-group` parameter (#27605)
* `az acr credential-set`: Add optional `--resource-group` parameter (#27605)
```

**Review by zhoxing-ms — APPROVED** [review]

**Review by evelyn-ys — DISMISSED** [review]

**Review by wangzelin007 — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,215 @@
...
+
+**Storage**
+
+* `az storage file/directory`: Add `--disallow-trailing-dot` (#27622)
+* `az storage share list-handle/close-handle`: Add `--disallow-trailing-dot` (#27622)
+* `az storage file copy start/start-batch`: Add `--disallow-source-trailing-dot` (#27622)
+* Fix #27590: `az storage fs directory download`: Check user sytem PATH for azcopy and use CLI config directory for new install (#27593)
+* `az storage account blob-inventory-policy create`: Add support for new filter `creationTime.lastNDays` in json (#27666)
+* `az storage account migration start/show`: Support customer inititated migration between replication options (#27692)
+
```
Comment (calvinhzy) [inline src/azure-cli/HISTORY.rst:214]:
```suggestion
* [BREAKING CHANGE] `az storage container-rm update`: Remove `--default-encryption-scope` and `--deny-encryption-scope-override` as they should only be specified during create (#27791)
```

**Review by bebound — APPROVED** [review]

**Review by calvinhzy — APPROVED** [review]

**Pull request merged**